source/Irrlicht/CBAWMeshFileLoader.h

```diff
@@ -20,16 +20,18 @@ namespace irr { namespace scene
 class CBAWMeshFileLoader : public IMeshLoader
 {
 private:
-	struct SPair
+	struct SBlobData
 	{
-		core::BlobHeaderV1* header;
-		void* blob;
+		const core::BlobHeaderV1* header;
+		const void* blob;
+		bool wasCreated;
```

Comment:
If you already have a `createdObjects` member in the context, then there is no need for a wasCreated flag.

Reply:
Thanks to the flag I don't need to look for an object in map to determine if it was loaded or not

Reply:
yes, and if the createdObj pointer is not NULL then it was definitely loaded.

You could also set it to 0xdeadbeefDEADBEEFull for things that failed to load.

Reply:
The whole point of my suggestions is that there is no 2nd map

Reply:
'fixed'